File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ['0' for _ in range(len(food) * 2 + 1)]
    
    for i, v in enumerate(food):
        a = str(i) * (v // 2)
        answer[i] = a
        answer[-1-i] = a
    
    return ''.join(answer)

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        StringBuilder sb = new StringBuilder("0");
        
        for (int i=food.length - 1; i>0; i--) {
            int share = food[i] / 2;
            for (int j=0; j<share; j++) {
                sb.insert(0, Integer.toString(i));
                sb.append(Integer.toString(i));
            }
        }
        
        return sb.toString();
    }
}